personal_budget_main.py

```diff
@@ -0,0 +1,214 @@
+import sys, os, warnings, getopt
+import datetime, string, random
+import openpyxl
+from tabulate import tabulate
+
+
+def extract_data_from_source(dir_path):
+    """
+    Extracts transaction data from files in source directory and returns it as dictionaries.
+    """
+    
+    transactions_all, monthly_summary  = {}, {}
+    
+    # Search for Excel files in source directory.
+    try:
+        excel_file_exist = False
+        file_list = os.listdir(dir_path)
+        for file_name in file_list:
+            if file_name.endswith(".xls") or file_name.endswith(".xlsx"):
+                excel_file_exist = True
+                file_url = dir_path + "/" + file_name
+
+                # Importing Excel data.
+                warnings.filterwarnings("ignore", ".*DrawingML*.")
+                try:
+                    wb = openpyxl.load_workbook(file_url)
+                    ws = wb.active
+                except Exception as e:
+                    print(f"Error: {e}")
+                    sys.exit(1)
```

Comment:
`sys.exit` is not used in Python programs. An error should be returned from here, or an exception should be raised. And the calling function should then act on the error or exception. Replace all sys.exit() with the same. 

Another approach is to return transactions_all, monthly_summary as empty dicts on this line, and then the caller function can terminate knowing that there were no transactions found